bin/thumbnail

```diff
@@ -391,10 +433,9 @@ function thumb_params() {
 			echo "Height ${origheight} x Width ${origwidth}"
 			if [[ $origheight -gt $HEIGHT && $origwidth -gt $WIDTH ]]; then
 				THUMB_PARAMS="${QUALITY} -thumbnail ${WIDTH}x${HEIGHT}+0+0^ -gravity center -crop ${WIDTH}x${HEIGHT}+0+0!"
-			elif [ $origheight -gt $HEIGHT ] || [ $origwidth -gt $WIDTH ]; then
-				THUMB_PARAMS="${QUALITY} -gravity center -crop ${WIDTH}x${HEIGHT}+0+0! -thumbnail ${WIDTH}x${HEIGHT}+0+0"
 			else
-				THUMB_PARAMS="${QUALITY}"
+				recalculate_requested_size_for_crop_down $origwidth $origheight
+				THUMB_PARAMS="${QUALITY} -gravity center -crop ${WIDTH}x${HEIGHT}+0+0 -thumbnail ${WIDTH}x${HEIGHT}+0+0"
```

Comment:
The main reason for undesired behavior of thumbnail script was not keeping an aspect ratio. While scaling is well-supported in imagemagick I couldn't find a way to request a crop operation that would give us what we wanted here *without* upscaling the image which is not a desired side-effect in a "down" mode. The main issue I don't know how to solve is that your highest priority is keeping the aspect ratio of the generated image in line with the requested aspect ratio and I couldn't find any working way to specify the aspect ratio in imagemagick's `geometry`.

http://www.imagemagick.org/script/command-line-processing.php#geometry

I've spent some time on finding a working solution but failed and fell back to do a workaround using ``bc`` @drsnyder  Do you want to give it a try?